File: problem-3/main.js
function largest_prime_factor(n){
    var i = 2
    while (i*i <= n) {
        if (n % i != 0){
            i++
        }
        else {
            n /= i
        }
    }
    return n
}
console.log(largest_prime_factor(600851475143))

File: problem-3/main.lua
function LargestPrimeFactor (n)
    local i = 2
    while i*i <= n do
        if n % i ~= 0 then
            i = i + 1
        else
            n = n / i
        end
    end
    return n
end
print(LargestPrimeFactor(600851475143))